phi.go

```diff
@@ -1,90 +1,156 @@
 package phi
 
-import "context"
+import (
+	"context"
+)
 
+// Message represents a message that is sent between tasks for communication.
+// It is an enum style interface where a struct can be made a variant of the
+// enum by implementing the interface.
 type Message interface {
 	IsMessage()
 }
 
-type MessageSync struct {
-	message   Message
-	responder chan Message
+// messageWithResponder is a `Message` wrapper that also contains a channel
+// where a response can be written.
+type messageWithResponder struct {
+	// The message being sent.
+	message Message
+
+	// The responder channel where the task will write its response.
+	responder chan Messages
 }
 
-type MessageBatch struct {
+// Messages is a collection of messages. Note that reducers will never receive
+// a `Messages` type because they will always be flattened before being
+// processed. If a task needs to respond with more than one message, and it is
+// important that these messages are processed together, then a custom
+// container type should be created and handled appropriately in the reducer.
+type Messages struct {
```

Comment:
I think this should be defined as `type Messages []Message` to prevent having an inner field with the same name as the struct.